src/nti/site/site.py

```diff
@@ -271,13 +324,16 @@ def _check_and_btree_maps(self, byorder):
 
     def changed(self, originally_changed):
         # If we changed, check and migrate
+
         if originally_changed is self:
             if (not isinstance(self._provided, self.btree_family.OI.BTree)
-                and len(self._provided) > self.btree_provided_threshold):
+                    and len(self._provided) >= self.btree_provided_threshold):
```

Comment:
We perform this check as the object is changed. Are we potentially re-ifying the whole structure to perform this check? If so, would we be better off doing a one-time migration on the `subscribers` data structure?

Reply:
> Are we potentially re-ifying the whole structure to perform this check?

Not exactly, we're pretty careful about that. We only look at objects that aren't BTrees already (dicts), and by definition those whole objects are already in memory. The keys are strings or classes, so those are also in memory, and copying that dict into a BTree will never need to touch the values (it will only have to sort the keys which are in memory).

Now, the `for iface, registrations in mapping.items()`, traversing the first level of the tree, *will* have to bring the *structure* into memory. But (a) we don't descend into it unless necessary and (b) right now it's always in memory for every read and write transaction; this will make it so that it only has to all be pulled in on write transactions. Hopefully that's an improvement. For many sites, that level of the tree is small anyway (e.g. in one of the sites we were looking at, there were only 13 entries at that level, which fits into a single bucket, and, in fact, wouldn't actually be turned into a BTree anyway, so no change there). 

>  If so, would we be better off doing a one-time migration on the subscribers data structure?

In general we can't. Adding and removing values will result in new plain dicts and tuples being put into the data structures; removing all utilities of a particular kind will even delete its entire part of the datastructure tree and adding a new one back will create it fresh. The super class doesn't have the hooks that would let us control what kind of objects it creates, so we have to do it after the fact. (Hmm, I should see about getting those hooks added.)

Alternately (I left a comment about this) we could use some fancy data structures that override methods like `__setitem__` and `__radd__` but hopefully that's overkill.